api/turing/cluster/virtual_service.go

```diff
@@ -29,16 +29,20 @@ func (cfg VirtualService) BuildVirtualService() *v1beta1.VirtualService {
 		},
 		Weight: 100,
 	}
-	httpMatches := make([]*networking.HTTPMatchRequest, len(cfg.MatchURIPrefixes))
-	for index, prefix := range cfg.MatchURIPrefixes {
-		uri := networking.HTTPMatchRequest{
-			Uri: &networking.StringMatch{
-				MatchType: &networking.StringMatch_Prefix{
-					Prefix: prefix,
+
+	var httpMatches []*networking.HTTPMatchRequest
+	if len(cfg.MatchURIPrefixes) > 0 {
+		httpMatches = make([]*networking.HTTPMatchRequest, len(cfg.MatchURIPrefixes))
```

Comment:
These additional lines are needed because newer versions of the Istio client specify `nil` in the `Spec.Http.[*].Match` field if there are no (`len(cfg.MatchURIPrefixes)==0`) http matches specified, instead of specifying a 0-length slice.